src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -25,7 +25,8 @@ use std::path::{Path, PathBuf};
 use std::str::{self, FromStr};
 use std::sync::Arc;
 
-/// Information about the platform target gleaned from querying rustc.
+/// Information about the platform target gleaned from querying rustc and from
+/// merging configs.
```

Comment:
This comment didn't really fit with this struct containing `rustflags` (which predates me working on target-applies-to-host). Putting `linker` in it makes the mismatch slightly worse, so I thought I'd update the comment.

I'm not entirely thrilled with the phrasing. If someone has a suggestion for better phrasing I'd be happy to take it.